js/renovation/button.tsx

```diff
@@ -16,6 +16,16 @@ const getImageContainerJSX = (source: string, position: string) => {
     }
 };
 
+const CustomRender = ({ text, icon, renderFn, contentRef, children }) => {
+    const result = renderFn({
+        model: { text, icon },
+    });
```

Comment:
I think it is a bad name - CustomRender

Reply:
> I think it is a bad name - CustomRender

I think a better name will be the `Template`. But the generator has a bug with translation all `template` words into `render` in the declaration.

Reply:
> Why have we created a model object?

It can be passed from 0 to 4 properties into the template.
We pass them using spread:
```jsx
// params = { model, index, onRendered, parent }
const data = { container: getPublicElement($container), ...params };
```